src/cunumeric/set/unique.cu

```diff
@@ -125,6 +187,24 @@ static Piece<VAL> tree_reduce(Array& output,
                                  cudaMemcpyDeviceToDevice,
                                  stream));
       merged.destroy();
+
+      if (return_index) {
+        indices.first.destroy();
+        other_index.first.destroy();
```

Comment:
Should you be early releasing this above like `other_piece`?

Reply:
I originally wanted to destroy the buffers here largely for the sake of clean code to minimize the number of special blocks  for the return_index case. Are you saying we should free these buffers closer to where we release my_piece and other_piece to potentially free up space before 
my_piece.first = output.create_output_buffer<VAL, 1>(buf_size); 
in line 182?